packages/contracts/contracts/test/Colosseum.t.sol

```diff
@@ -217,179 +239,261 @@ contract Colosseum_Test is Colosseum_Initializer {
+        assertEq(colosseum.isInProgress(outputIndex), true);
 
-        _proveFault();
+        _proveFault(outputIndex);
     }
 
-    function test_timeoutWhenTimedOutAsserterTurn() external {
-        _createChallenge();
-        Types.Challenge memory challenge = colosseum.getChallengeInProgress();
+    function test_proveFault_whenAsserterTimedOut_succeeds() external {
+        uint256 outputIndex = oracle.latestOutputIndex();
+        _createChallenge(outputIndex);
+        Types.Challenge memory challenge = colosseum.getChallenge(outputIndex);
 
-        assertEq(colosseum.isInProgress(), true);
-        assertEq(challenge.next, asserter);
+        assertEq(colosseum.isInProgress(outputIndex), true);
+        assertEq(nextSender(challenge), challenge.asserter);
 
         vm.warp(challenge.timeoutAt + 1);
         // check the asserter timeout
-        assertEq(colosseum.isInProgress(), true);
+        assertEq(colosseum.isInProgress(outputIndex), true);
         assertEq(
-            uint256(colosseum.getStatusInProgress()),
+            uint256(colosseum.getStatus(outputIndex)),
             uint256(Colosseum.ChallengeStatus.ASSERTER_TIMEOUT)
         );
 
-        vm.prank(challenger);
-        colosseum.asserterTimeout();
+        _proveFault(outputIndex);
+    }
+
+    function test_approveChallenge_succeeds() public {
+        test_proveFault_succeeds();
+
+        uint256 outputIndex = oracle.latestOutputIndex();
+        Types.CheckpointOutput memory prevOutput = oracle.getL2Output(outputIndex);
+
+        vm.prank(guardian);
+        colosseum.approveChallenge(outputIndex);
+
+        Types.CheckpointOutput memory newOutput = oracle.getL2Output(outputIndex);
+
+        assertTrue(prevOutput.outputRoot != newOutput.outputRoot);
+        assertEq(prevOutput.timestamp, newOutput.timestamp);
+        assertEq(prevOutput.l2BlockNumber, newOutput.l2BlockNumber);
+        assertEq(newOutput.submitter, challenger);
+        assertEq(outputIndex, oracle.latestOutputIndex());
+    }
+
+    function test_approveChallenge_notGuardian_reverts() external {
+        test_proveFault_succeeds();
+
+        uint256 outputIndex = oracle.latestOutputIndex();
+
+        vm.prank(trusted);
+        vm.expectRevert("Colosseum: sender is not the guardian");
+        colosseum.approveChallenge(outputIndex);
     }
 
-    function test_timeoutWhenTimedOutChallengerTurn() external {
-        uint256 challengeId = _createChallenge();
-        Types.Challenge memory challenge = colosseum.getChallengeInProgress();
+    function test_approveChallenge_notProven_reverts() external {
+        uint256 outputIndex = oracle.latestOutputIndex();
+
+        vm.prank(guardian);
+        vm.expectRevert("Colosseum: this challenge is not proven");
+        colosseum.approveChallenge(outputIndex);
+    }
 
-        assertEq(colosseum.isInProgress(), true);
-        assertEq(challenge.next, asserter);
+    function test_challengerTimeout_succeeds() external {
```

Comment:
When challenger timed out, a new challenger can be created again. Why don't we cover this case that an asserter can create a new one with doubled bond?